substrate/fd

```diff
@@ -70,41 +76,36 @@ namespace substrate
 			return *this;
 		}
 
-		SUBSTRATE_NO_DISCARD(operator int32_t() const noexcept) { return fd; }
-		SUBSTRATE_NO_DISCARD(bool operator ==(const int32_t desc) const noexcept) { return fd == desc; }
-		SUBSTRATE_NO_DISCARD(bool valid() const noexcept) { return fd != -1; }
-		SUBSTRATE_NO_DISCARD(bool isEOF() const noexcept) { return eof; }
-		void invalidate() noexcept { fd = -1; }
+		SUBSTRATE_NO_DISCARD(inline operator int32_t() const noexcept) { return fd; }
```

Comment:
These should be implicitly `inline` according to [the draft standard](https://eel.is/c++draft/class.mfct#1) so do not require explicitly marking.

Reply:
I've added further notes to the last commit in the set. Basically, while the Standard says so, Microsoft says *nyet*: https://learn.microsoft.com/en-us/cpp/build/reference/zc-inline-remove-unreferenced-comdat?view=msvc-170